packages/jobs/lib/integration.service.ts

```diff
@@ -10,18 +10,54 @@ import {
     isProd,
     ServiceResponse,
     NangoError,
-    formatScriptError
+    formatScriptError,
+    isOk
 } from '@nangohq/shared';
-import { getOrStartRunner, getRunnerId } from './runner/runner.js';
+import { Runner, getOrStartRunner, getRunnerId } from './runner/runner.js';
 import tracer from './tracer.js';
 
+interface ScriptObject {
+    context: Context | null;
+    runner: Runner;
+    activityLogId: number | undefined;
+}
+
 class IntegrationService implements IntegrationServiceInterface {
-    public runningScripts: { [key: string]: Context } = {};
+    public runningScripts: Map<string, ScriptObject>;
 
     constructor() {
+        this.runningScripts = new Map();
         this.sendHeartbeat();
     }
 
+    async cancelScript(syncId: string, environmentId: number): Promise<void> {
+        const scriptObject = this.runningScripts.get(syncId);
+
+        if (!scriptObject) {
+            return;
+        }
+
+        const { runner, activityLogId } = scriptObject;
+
+        const res = await runner.client.cancel.mutate({
+            syncId
+        });
+
+        if (isOk(res)) {
+            this.runningScripts.delete(syncId);
+        } else {
+            if (activityLogId && environmentId) {
+                await createActivityLogMessage({
+                    level: 'error',
+                    environment_id: environmentId as number,
+                    activity_log_id: activityLogId as number,
+                    content: `Failed to cancel script`,
+                    timestamp: Date.now()
+                });
+            }
```

Comment:
I think we the if you are covered already
```suggestion
            if (activityLogId && environmentId) {
                await createActivityLogMessage({
                    level: 'error',
                    environment_id: environmentId,
                    activity_log_id: activityLogId,
                    content: `Failed to cancel script`,
                    timestamp: Date.now()
                });
            }
```

Reply:
Thanks